libbeat/template/config.go

```diff
@@ -32,6 +32,8 @@ type TemplateConfig struct {
 	AppendFields common.Fields    `config:"append_fields"`
 	Overwrite    bool             `config:"overwrite"`
 	Settings     TemplateSettings `config:"settings"`
+	// not configurable by the user
+	Product string
```

Comment:
I think we should be more generic and instead have [`Meta`](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-meta-field.html) field that take a `common.MapStr` so if we ever need to add more fields to the template it will be more flexible.

Also, we could expose that to the user, I do not see a problem with that, on your side you can scope them with a prefix and merged them with what the user has defined.

Reply:
the meta field is already added internally with the version here: https://github.com/elastic/beats/pull/10313/files#diff-2a54812d26ee6b4a1d05d8dcd26d07a6R225

if I add it as a `MapStr` and expose it to the users, it means that they will be free to overwrite the version: is that what you want?

alternatively define a field (pick a name) as a `MapStr` to insert inside `_meta`?

Reply:
I've look more on the existing code, I think we were both doing something we do not need to do.

You had some concerns about exposing the _meta field to the user and the risk of allowing people to change it.

My principal concern with the proposal was the addition of the `Product` field in the `TemplateConfig` struct, but if you look at the [constructor of a template](https://github.com/elastic/beats/blob/master/libbeat/template/template.go#L55) we already receive the beat name (which is the _product_)

We could add the `beatName` on the template struct and use `beat_name` in the meta field, for me the change look minimal and should in theory works as is with your implementation without changing anything.

Reply:
That is a good & simple idea! Changed it now, mind have another look? (I force-pushed because the previous code didn't make sense any longer, i hope that's ok)